README.md

```diff
@@ -21,11 +21,27 @@ You can find the installation documentation for the
 The documentation for advanced usage of Jupyter notebook can be found
 [here](https://jupyter-server.readthedocs.io/en/latest/).
 
-For a local installation, make sure you have
+To install the latest release locally, make sure you have
 [pip installed](https://pip.readthedocs.io/en/stable/installing/) and run:
 
     $ pip install jupyter_server
 
+### Versioning and Branches
+
+If Jupyter Server is a dependency of your project/application, it is important that you pin it to a version that works for your application. Currently, Jupyter Server only has minor versions and patch versions. Different minor versions may include API-changes while patch versions do not change API.
+
+Each minor version has its own branch in this repo, and patches are applied to their relevant minor version. The master branch of the latest minor version.
+
+The available minor versions are:
+* [0.2.x](https://github.com/jupyter/jupyter_server/tree/0.2.x): Extension Application API added.
+* [0.3.x](https://github.com/jupyter/jupyter_server/tree/master) (master): Kernels API changed. Kernel Providers are the new way to interface with kernels.
```

Comment:
If retained, I'd prefer slightly different wording of _"Kernels API changed"_ to avoid ambiguity.  How about _"Kernel management framework has been replaced"_?